src/internal/types.ts

```diff
@@ -31,6 +31,12 @@ export interface OperatorFunction<T, R> extends UnaryFunction<Observable<T>, Obs
 
 export type FactoryOrValue<T> = T | (() => T);
 
+/**
+ * A function type interface that describes a function that accepts and returns a parameter of the same type.
+ *
+ * It only works with a single type of data. Input and output value must be of the same type.
```

Comment:
I'd like the docs to mention `OperatorFunction`since `MonoTypeOperatorFunction` is used to describe it. Actually, I'm not sure how much this sentence is useful. Maybe replace it with something like this:

```suggestion
 * Used to describe {@link OperatorFunction} with the only one type: `OperatorFunction<T, T>`.
```

This is just suggestion, maybe there's a better way to do it.